content-services/content-sources-backend
# Fixes 4825: add template attribute for when env is created

## Summary

adds an rhsm_environment_created db column and api attribute.  It is false when a template is first created.  Once it is created via the update-template-content task, it gets set to true.

## Testing steps

1.  create and snapshot one repository
2.  Create a template with the repository:
```
####
POST http://localhost:8000/api/content-sources/v1.0/templates/
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
Content-Type: application/json

{
  "name":"test7",
  "arch": "x86_64",
  "version": "9",
  "repository_uuids": ["270fabbd-ca28-4cd0-9cff-a274b4e797b9"],
  "date": "2024-09-10T15:09:43-04:00"
}
```

3.  check teh return:

  "rhsm_environment_created": false

4.  fetch the created template:
```
###
GET http://localhost:8000/api/content-sources/v1.0/templates/687c93e2-b146-428d-b65d-a972bed5edbe
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
//x-Rh-Insights-Request-Id: 9d229d34-7219-405d-ba3d-8f99cf7c36ae
Content-Type: application/json
```

after a few seconds of creation, it should now be true:
```
  "rhsm_environment_created": true
```
